### ISSUE TITLE
GameGuru MAX - Characters, Creatures, and Animation???

### ISSUE THREAD

**Issue by MonkeyFrogStudio** [TheGameCreators/GameGuruRepo#3997]
I am just sort of thinking here, seeking answers, but in the form of a request (possibly). 

In the "old days" ... at least with other game engines ... you could import any animated model and it didn't matter how it was animated (bones, purely keyframe, etc.), as long as each animation had the correct name, it would get called, and would work. Sort of like how the door animations work in MAX now. As long as you have Open and Close, the Behavior will call them and they will work.

So, then, why doesn't this same thing work with NPCs? Why do we have to go through the hassle of trying to have a rig that matches what MAX is using, with all the correct bone names, etc.? Why can't we animate a character however we want, using whatever method we want, bring that animated model into MAX, and have it just ... work? 

In other words, what if I want to create a creature that's not even remotely human, but want to use it as an enemy. Why can't it have an "attack" animation, "move" animation, "death" animation, etc., and as long as they are named appropriately, they will get called, the animation play, and it just works?

I realize the current system is for us to get rigged humanoid characters into MAX and then let MAX animate them with the animations MAX has. Fine. But what about non-humanoid creatures (dragons, wolves, strange carnivorous plants, etc.)? How will we use them in MAX so they can patrol, sense the player (and each other), attack, etc.?

**Comment by Granada1** [comment]
That’s a good idea , automatically write the FPE file for the imported model . If I’m reading you right !

**Comment by MonkeyFrogStudio** [comment]
Well, however it would be done, I don't know. ;) But when I worked in game engines in the past, they only required you to either name your animations properly to work or you had to create the animation names in the engine and ensure they matched the animations of your models. So, if you wanted a model to "walk" you had to name your animation "walk", "run" to "run", etc. It didn't matter if the model had two legs or eight, if the model used bones or keyframed animations. If the name matched, it played that animation. It was simple to use and required nothing special to get an animated model into and working in the engine.

**Comment by davetgc** [comment]
I'm going to mark this as an enhancement, but I agree, it is a good idea.
